scripts/run_monai_unet.py

```diff
@@ -1,20 +1,58 @@
-from monai.config import print_config
-import os
+import json
+import inspect
 import typer
 from pathlib import Path
 from typing import List
 
 from segmantic.prepro.labels import load_tissue_list
 from segmantic.seg import monai_unet
 
+app = typer.Typer()
+
 
 def get_nifti_files(dir: Path) -> List[Path]:
     if not dir:
         return []
     return sorted([f for f in dir.glob("*.nii.gz")])
 
 
-def main(
+@app.command()
+def train_config(
+    config_file: Path = typer.Option(
+        ..., "--config-file", "-c", help="config file in json format"
+    ),
+    print_defaults: bool = False,
+):
+    """Train UNet with configuration provided as a json file
+
+    Example invocation:
+
+        --config-file my_config.json
+
+    To generate a default config:
+
+        --config-file my_config.json --print-defaults
+    """
+    if print_defaults:
+        sig = inspect.signature(monai_unet.train)
+
+        default_args = {
+            k: v.default
+            if v.default is not inspect.Parameter.empty
+            else f"<required option: {v.annotation.__name__}>"
+            for k, v in sig.parameters.items()
+        }
+        with open(config_file, "w") as f:
+            json.dump(default_args, f, indent=4)
```

Comment:
@pcrespov this fails if the default is a pathlib.Path, because it is not json serializable. Any tips?

Reply:
Create your own ``json_dump`` function as i did [here](https://github.com/ITISFoundation/osparc-simcore/blob/a50b61735381231abba2cfcd57f3314785c656b0/packages/service-library/src/servicelib/json_serialization.py#L7). 

And here in one liner :-)
```python
# dump
config_file.write_text( json_dumps(default_args, indent=4) )

# load
monai_unet.train(** json.loads(config_file.read_text()) )
```

NOTE: Notice ``S`` in load and dump .. which refers to string

Reply:
cool. thanks. I must check out the rest of the pathlib.Path API